Automattic/wp-calypso
# Jetpack Onboarding: Fix user hash for connected sites

Currently, when tracking events in the Jetpack Onboarding flow, we use the onboarding email. However, when the site gets connected, we don't use the onboarding credentials anymore, and the onboarding email is no longer there. This causes issues with determining the user hash for connected sites.

This PR updates the selector to also account for connected sites. It renames the selector to reflect the new behavior, and updates tests to account for that.

Fixes #23297.

To test:
* Checkout this branch
* Grab a non-connected site where you're logged in with a user with the same email as your .com user.
* Start the onboarding flow at http://yourgroovydomain.com/wp-admin/admin.php?page=jetpack&action=onboard&calypso_env=development where `yourgroovydomain.com` is your site.
* In the console in Calypso, type `localStorage.setItem( 'debug', 'calypso:analytics:tracks' )`.
* Fill something in the site title field and submit the form.
* Verify the `calypso_jpo_site_title_submitted` event is recorded properly with a `user_id` that corresponds to `jpo_user_SOME_HASH_HERE`.
* Grab a connected site.
* Start the onboarding flow at http://yourgroovydomain.com/wp-admin/admin.php?page=jetpack&action=onboard&calypso_env=development where `yourgroovydomain.com` is your site.
* In the console in Calypso, type `localStorage.setItem( 'debug', 'calypso:analytics:tracks' )` if necessary.
* Verify the `calypso_jpo_site_title_submitted` event is recorded properly with a `user_id` that corresponds to `jpo_user_SOME_HASH_HERE`.
* Verify the `SOME_HASH_HERE` is the same for those 2 test cases.
